javav2/example_code/ecr/src/main/java/com/example/ecr/scenario/ECRScenario.java

```diff
@@ -0,0 +1,254 @@
+        the security and integrity of your container images. The repository policy allows you to 
+        define specific rules and restrictions for accessing and managing the images stored within your ECR 
+        repository.    
+        """);
+        waitForInputToContinue(scanner);
+        ecrActions.setRepoPolicy(repoName, iamRole);
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("""
+        3. Display ECR repository policy.
+       
+        Now we will retrieve the ECR policy to ensure it was successfully set.   
+        """);
+        waitForInputToContinue(scanner);
+        String policyText = ecrActions.getRepoPolicy(repoName);
+        System.out.println("Policy Text:");
+        System.out.println(policyText);
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("""
+        4. Retrieve an ECR authorization token.
+       
+        You need an authorization token to securely access and interact with the Amazon ECR registry. 
+        The `getAuthorizationToken` method of the `EcrAsyncClient` is responsible for securely accessing 
+        and interacting with an Amazon ECR repository. This operation is responsible for obtaining a 
+        valid authorization token, which is required to authenticate your requests to the ECR service. 
+        
+        Without a valid authorization token, you would not be able to perform any operations on the 
+        ECR repository, such as pushing, pulling, or managing your Docker images.    
+        """);
+        waitForInputToContinue(scanner);
+        ecrActions.getAuthToken();
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("""
+        5. Get the ECR Repository URI.
+                    
+        The URI  of an Amazon ECR repository is important. When you want to deploy a container image to 
+        a container orchestration platform like Amazon Elastic Kubernetes Service (EKS) 
+        or Amazon Elastic Container Service (ECS), you need to specify the full image URI, 
+        which includes the ECR repository URI. This allows the container runtime to pull the 
+        correct container image from the ECR repository.    
+       """);
+        waitForInputToContinue(scanner);
+        String repositoryURI = ecrActions.getRepositoryURI(repoName);
+        System.out.println("The repository URI is "+repositoryURI);
+        waitForInputToContinue(scanner);
+
+        System.out.println(DASHES);
+        System.out.println("""
+        6. Set an ECR Lifecycle Policy.
+                    
+        An ECR Lifecycle Policy is used to manage the lifecycle of Docker images stored in your ECR repositories. 
+        These policies allow you to automatically remove old or unused Docker images from your repositories, 
+        freeing up storage space and reducing costs.    
```

Comment:
What does this specific policy do? Might want to let the user know.

Reply:
Nice comment. I will add this info